src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -228,8 +228,8 @@ internal override NamespaceExtent Extent
                     // NOTE: the following is not cancellable.  Once we've set the
                     // members, we *must* do the following to make sure we're in a consistent state.
                     this.DeclaringCompilation.DeclarationDiagnostics.AddRange(diagnostics);
-
                     RegisterDeclaredCorTypes();
+                    DeclaringCompilation.SymbolDeclaredEvent(this);
```

Comment:
Please add a comment clearly stating when do we intend to produce this event so that it would be clear why this place is the right point to do this, why not earlier or later than that.